tpm/aziot-tpm-sys/azure-iot-hsm-c/src/hsm_log.c

```diff
@@ -25,18 +23,20 @@ void log_init(int level) {
 
 void log_msg(int level, const char* file, const char* function, int line, const char* fmt_str, ...) {
     static char levels[3][5] = {"DBUG", "INFO", "ERR!"};
-    static int  syslog_levels[3] = { 7, 6, 3 };
+    static int syslog_levels[3] = { 7, 6, 3 };
 
     if (level >= log_level) {
         time_t now;
-        char buffer[MAX_LOG_SIZE];
         char time_buf[sizeof("2018-05-24T00:00:00Z")];
         time(&now);
         strftime(time_buf, sizeof(time_buf), "%FT%TZ", gmtime(&now));
+        fprintf(stderr, "<%d>%s [%s] (%s:%s:%d) ", syslog_levels[level], time_buf, levels[level], file, function, line);
+
         va_list args;
         va_start (args, fmt_str);
-        vsnprintf(buffer, MAX_LOG_SIZE, fmt_str, args);
```

Comment:
What was the original purpose of this buffering?  I have not seen this style of logging before.

Reply:
It's [ancient history](https://github.com/Azure/iotedge/commit/300b23c7672db7530da1eea92dca369590a4510c#diff-566e03479a0e9844cae17e0b11682147a847f45b8e69bffbd507735bc0eb1a35R21) at this point.